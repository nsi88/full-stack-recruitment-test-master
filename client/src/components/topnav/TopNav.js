import React from 'react';
import './TopNav.scss';
import logo from '../../icons/logo.svg';
import burgerMenu from '../../icons/burger-menu.svg';

const TopNav = () => (
  <nav className='top-nav'>
    <a href="/">
      <span className='top-nav__logo-text'>Skyscanner</span>
      <img className='top-nav__logo' alt="Skyscanner" src={logo}/>
    </a>
    <img className='top-nav__burger-menu' alt='Menu' src={burgerMenu}/>
  </nav>
);

export default TopNav;
