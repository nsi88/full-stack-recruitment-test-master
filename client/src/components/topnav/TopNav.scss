@import "~bpk-mixins";

.topNav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: $bpk-spacing-sm $bpk-spacing-sm;

  background-color: $bpk-color-white;
  box-shadow: 0 0 10px 0 $bpk-color-gray-300;
}

.logo {
  vertical-align: middle;
}

.logoText {
  // REVIEW: Do I need to show it?
  @include bpk-visually-hidden;
}

.burgerMenu {
  vertical-align: middle;
  width: $bpk-icon-size-lg;
  height: $bpk-icon-size-lg;
  cursor: pointer;
}
