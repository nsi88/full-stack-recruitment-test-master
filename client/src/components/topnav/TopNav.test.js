import React from 'react';
import TopNav from '../TopNav';
import renderer from 'react-test-renderer';

test('render', () => {
  const tree = renderer.create(<TopNav/>).toJSON();
  expect(tree).toMatchSnapshot();
});
