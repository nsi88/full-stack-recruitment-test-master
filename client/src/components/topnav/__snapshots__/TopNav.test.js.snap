// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<nav
  className="top-nav"
>
  <a
    href="/"
  >
    <span
      className="top-nav__logo-text"
    >
      Skyscanner
    </span>
    <img
      alt="Skyscanner"
      className="top-nav__logo"
      src="test-file-stub"
    />
  </a>
  <img
    alt="Menu"
    className="top-nav__burger-menu"
    src="test-file-stub"
  />
</nav>
`;
